data/Builder.go

```diff
@@ -0,0 +1,72 @@
+package data
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tidepool-org/platform/validate"
+)
+
+type Datum map[string]interface{}
+
+type DatumArray []Datum
+
+type Builder interface {
+	BuildFromDatum(datum Datum) interface{}
```

Comment:
What is the return value `interface{}` represent?  Should we have a type for it so the function signature is clear?